test/kubemark/resources/start-kubemark-master.sh

```diff
@@ -241,6 +241,52 @@ function prepare-log-file {
 	chown root:root $1
 }
 
+# A helper function for removing salt configuration and comments from a file.
+# This is mainly for preparing a manifest file.
+#
+# $1: Full path of the file to manipulate
+function remove-salt-config-comments {
+	# Remove salt configuration.
+	sed -i "/^[ |\t]*{[#|%]/d" $1
+	# Remove comments.
+	sed -i "/^[ |\t]*#/d" $1
+}
+
+# Prepare the abac policy file from the salt template.
+function setup-abac-authz-policy-file {
```

Comment:
> Again, I didn't follow carefully what was happening in this ares, so @deads2k please correct me if I'm saying something stupid, but I think we should just make our kubemark components behave as the corresponding roles in the real cluster.

I think you are correct about the hollowKubelet and the controller manager.

The `kubectl` user though.  Is it just a scripting user you use to support/run your tests?   It's ok to grant wide permissions to the bits that run or bootstrap your CI and then create custom roles if you need to.  I wouldn't want to have CI related roles in the default roles we create.  In openshift, we run a lot of the CI bootstrapping/housekeeping as a cluster-admin and all the user level tests as unprivileged users.